backend/tarsy/services/parallel_stage_executor.py

```
@@ -0,0 +1,1005 @@
+            # ALL policy: all must succeed (any failure = stage failure)
+            return StageStatus.COMPLETED if failed_count == 0 else StageStatus.FAILED
+        else:  # FailurePolicy.ANY
+            # ANY policy: at least one must succeed (all failures = stage failure)
+            return StageStatus.COMPLETED if completed_count > 0 else StageStatus.FAILED
+    
+    async def _execute_parallel_stage(
+        self,
+        stage: "ChainStageConfigModel",
+        chain_context: ChainContext,
+        session_mcp_client: MCPClient,
+        chain_definition: "ChainConfigModel",
```

Ruff (0.14.8)

Unused method argument: `chain_definition`

(ARG002)